common/txmgr/types/mocks/utils.go

```diff
@@ -0,0 +1,15 @@
+package mocks
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/txmgr"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+)
+
+type MockEvmTxStore = TxStore[evmtypes.Address, *big.Int, evmtypes.TxHash, evmtypes.BlockHash, txmgr.EvmNewTx, *evmtypes.Receipt, txmgr.EvmTx, txmgr.EvmTxAttempt, evmtypes.Nonce]
+
+func NewMockEvmTxStore(t *testing.T) *MockEvmTxStore {
+	return NewTxStore[evmtypes.Address, *big.Int, evmtypes.TxHash, evmtypes.BlockHash, txmgr.EvmNewTx, *evmtypes.Receipt, txmgr.EvmTx, txmgr.EvmTxAttempt, evmtypes.Nonce](t)
```

Comment:
Do we have to list all of these types out individually? Or could we infer some of them? e.g. given a tx, doesn't that tell us tx_hash and sequence type implicitly?

Reply:
I think they would still all have to be listed out on the `type TxStore[...]`, but it should be possible to omit a handful of them at call sites, however we'd have to constrain the types more strictly - not sure if that is helpful or not.

Reply:
I didn't quite understand what changes you are suggesting.
If we have TxStore type using x number of generic types, then each call site has to specify the native types there, no?
How can we omit certain types?

Also, I didn't get what you mean by "infer some of them".

Reply:
They don't always need to be explicitly listed. You can often omit the whole set: https://go.dev/blog/intro-generics#type-inference

Reply:
This doesn't seem like a good case for it, but this collection of types are connected enough that there could be other opportunities.

Reply:
Ohk, get it now.
Yes, it can apply for cases which match this criterion << Function argument type inference only works for type parameters that are used in the function parameters.

We could use it when we have function params matching the function argument types.